maven-wrapper/src/main/java/org/apache/maven/wrapper/Installer.java

```diff
@@ -70,8 +70,12 @@ public Path createDist( WrapperConfiguration configuration )
         {
             Logger.info( "Detected MVNW_REPOURL environment variable " + mvnwRepoUrl );
             String mvnPath = distributionUrl.toURL().toString();
-            mvnPath = mvnPath.substring( mvnPath.indexOf( "org/apache/maven" ) );
-            distributionUrl = new URI( mvnwRepoUrl ).resolve( "/" ).resolve( mvnPath );
+            int pathIdx =  mvnPath.indexOf( "org/apache/maven" );
+            if ( 0 <= pathIdx )
+            {
+                mvnPath = mvnPath.substring( pathIdx );
+                distributionUrl = new URI( mvnwRepoUrl ).resolve( "/" ).resolve( mvnPath );
+            }
```

Comment:
How it is connected with new script type?

`pathIdx` will be `-1` if there is no substring ....